app/admin/quests/dashboard/[questId]/page.tsx

```diff
@@ -509,6 +522,15 @@ export default function Page({ params }: QuestIdProps) {
             cta: step.data.balance_cta,
             href: step.data.balance_href,
           });
+        } else if (step.type === "Contract") {
+          await AdminService.createContract({
+            quest_id: questId.current,
+            name: step.data.contract_name,
+            desc: step.data.contract_desc,
+            href: step.data.contract_href,
+            cta: step.data.contract_cta,
+            calls: step.data.calls,
+          });
         }
       } catch (error) {
```

Comment:
In this catch, you can add
```tsx
showNotification(`Error adding ${step.type} task: ${error}`, "error");
```
So that if there is any issue while parsing the JSON, the user is instantly informed